SPEC.md

```diff
@@ -0,0 +1,439 @@
+[Multi-Network CRD Specification](https://github.com/k8snetworkplumbingwg/multi-net-spec).
+
+# 1 Document Versioning
+
+This document shall be versioned in the MAJOR.MINOR.PATCH format. The
+specification version will be managed by GitHub tags with the form “v1.0.0” in
+this specification repository:
+
+https://github.com/k8snetworkplumbingwg/device-info-spec/ 
+
+The SPEC.md file is the definitive specification and any other formats (PDF,
+etc) are only provided for convenience.
+
+# 2 Definitions
+
+## 2.1 Container Network Interface (CNI)
+
+CNI consists of a specification and libraries for writing plugins for writing
+plugins to configure network interfaces in Linux containers.
+
+## 2.2 Device Plugin
+
+A framework in Kubernetes for resources to be advertised to Kubelet. A general
+rule is that Device Plugins manage limited resources.
+
+## 2.3 Workload
+
+Application running in a container.
+
+## 2.4 Network Attachment Definition
+
+Kubernetes allows custom resources to be created via a Custom Resource
+Definition (CRD). The Network Plumbing Working Group specification uses a CRD
+called a Network-Attachment-Definition to define a network, and configuration
+data about that network.
+
+## 2.5 Device
+
+A “device”, as referred to in this specification, is one of possible several
+interfaces being consumed by a container. A variety of technologies can be used
+to back the device.
+
+# 3 Device Information Format
+
+The Device Information Format is the json format that all Device Information
+files must follow. The fields are defined as follows.
+
+## 3.1 “device-info” keys
+
+The “device-info” map must contain the following keys.
+
+### 3.1.1 “type”
+
+This <ins>required</ins> key’s value (type string) shall indicate the type of
+device that is being described and determines what additional information is
+being included. This key’s value must contain one of the following values:
+
+- **“pci”**: Indicates that a PCI device is being described.
+- **“vdpa”**: Indicates that a vDPA device is being described.
+- **“vhost-user”**: Indicates that a vhost-user device is being described.
+- **“memif”**: Indicates that a memif device is being described.
+
+### 3.1.2 “version”
+
+This <ins>required</ins> key’s value (type string) must match the Device
+Information Specification version that is being used. The format of the string
+shall be “<MAJOR>.<MINOR>.<PATCH>” (i.e. “1.0.0”).
+
+### 3.1.3 “pci”
+
+This <ins>optional</ins> (<ins>required</ins> if “type” is set to “pci”) key’s
+value (type map) shall contain PCI device information. The map may contain the
+following keys.
+
+#### 3.1.3.1 “pci-address”
+
+This <ins>required</ins> key’s value (type string) shall contain the PCI Address
+of the device (in standard BDF format “dddd:BB:DD.f”).
+
+#### 3.1.3.2 “vhost-net”
+
+This <ins>optional</ins> key’s value (type string) shall contain the path to
+the vhost-net char device used as an exceptional path.
```

Comment:
I don't know if everyone would be familiar with this. If you think people might not understand this well, we could add a link to the guide: https://doc.dpdk.org/guides/howto/virtio_user_as_exceptional_path.html

Reply:
Good suggestion. Done.